modules/apps/portal-workflow/portal-workflow-kaleo-runtime-integration-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/integration/internal/WorkflowDefinitionManagerImpl.java

```diff
@@ -109,12 +112,26 @@ public List<WorkflowDefinition> getActiveWorkflowDefinitions(
 
 			serviceContext.setCompanyId(companyId);
 
-			List<KaleoDefinition> kaleoDefinitions =
-				_kaleoDefinitionLocalService.getScopeKaleoDefinitions(
-					WorkflowDefinitionConstants.SCOPE_ALL, true, start, end,
-					KaleoDefinitionOrderByComparator.getOrderByComparator(
-						orderByComparator, _kaleoWorkflowModelConverter),
-					serviceContext);
+			List<KaleoDefinition> kaleoDefinitions = null;
+
+			if (KaleoDefinitionThreadLocal.
+					isSkipKaleoDefinitionResourcePermission()) {
+
+				kaleoDefinitions =
+					_kaleoDefinitionLocalService.getScopeKaleoDefinitions(
+						WorkflowDefinitionConstants.SCOPE_ALL, true, start, end,
+						KaleoDefinitionOrderByComparator.getOrderByComparator(
+							orderByComparator, _kaleoWorkflowModelConverter),
+						serviceContext);
+			}
+			else {
+				kaleoDefinitions =
+					_kaleoDefinitionService.getScopeKaleoDefinitions(
+						WorkflowDefinitionConstants.SCOPE_ALL, true, start, end,
+						KaleoDefinitionOrderByComparator.getOrderByComparator(
+							orderByComparator, _kaleoWorkflowModelConverter),
+						serviceContext);
+			}
```

Comment:
Create a private method in this class that will remove the else statement and simplify the code, for example:

````java
private <T> T _get(
    UnsafeSupplier<T> localServiceSupplier,
    UnsafeSupplier<T> serviceSupplier) {

    return KaleoDefinitionResourcePermissionThreadLocal.isDisabled() ? 
                localServiceSupplier.get() : serviceSupplier.get();
}
````

so

````java
List<KaleoDefinition> kaleoDefinitions = _get(
    () -> _kaleoDefinitionLocalService.getScopeKaleoDefinitions(
		WorkflowDefinitionConstants.SCOPE_ALL, true, start, end,
		KaleoDefinitionOrderByComparator.getOrderByComparator(
			orderByComparator, _kaleoWorkflowModelConverter), serviceContext),
    () -> _kaleoDefinitionService.getScopeKaleoDefinitions(
		WorkflowDefinitionConstants.SCOPE_ALL, true, start, end,
		KaleoDefinitionOrderByComparator.getOrderByComparator(
			orderByComparator, _kaleoWorkflowModelConverter), serviceContext));
````